.github/workflows/lint.yml

```diff
@@ -0,0 +1,69 @@
+name: lint
+
+on: [push, pull_request]
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v1
+
+    - name: Set up python 3.8
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
+        cache: "pip"
+        cache-dependency-path: |
+          requirements_dev.txt
+
+    - name: Install requirements
+      run: python -m pip install -r requirements_dev.txt
+
+    - name: Run pre-commit
+      uses: pre-commit/action@v2.0.3
+
+
+  pyright:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.8", "3.9", "3.10"]
+      fail-fast: false
+    env:
+      PYRIGHT_VERSION: 1.1.221  # should match task.env
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v1
+
+    - name: Set up python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+        cache: "pip"
+        cache-dependency-path: |
+          requirements_dev.txt
+          setup.py
+
+    - name: Install requirements
+      run: python -m pip install -r requirements_dev.txt .[voice,docs,speed]
+
```

Comment:
About here we can use dotenv to get the environment variable out of task.env
``dotenv -f task.env get PYRIGHT_PYTHON_FORCE_VERSION``

Reply:
Yup, nice idea, ty
a5075b122804a8753038073b9202988dec55ee4f